modules/widgets/authors.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/*
+ * Widget to display blog authors with avatars and recent posts.
+ *
+ * Configurable parameters include:
+ * 1. Whether to display authors who haven't written any posts
+ * 2. The number of posts to be displayed per author (defaults to 0)
+ * 3. Avatar size
+ */
+class Widget_Authors extends WP_Widget {
+	public function __construct() {
+		parent::__construct(
+			'authors',
+			/** This filter is documented in modules/widgets/facebook-likebox.php */
+			apply_filters( 'jetpack_widget_name', __( 'Authors', 'jetpack' ) ),
+			array( 'classname' => 'widget_authors', 'description' => __( 'Display blogs authors with avatars and recent posts.', 'jetpack' ) ),
+			array( 'width' => 300 )
+		);
+
+		add_action( 'publish_post', array( __CLASS__, 'flush_cache' ) );
+		add_action( 'deleted_post', array( __CLASS__, 'flush_cache' ) );
+		add_action( 'switch_theme', array( __CLASS__, 'flush_cache' ) );
+	}
+
+	public static function flush_cache() {
+		wp_cache_delete( 'widget_authors', 'widget' );
+		wp_cache_delete( 'widget_authors_ssl', 'widget' );
+	}
+
+	public function widget( $args, $instance ) {
+		global $wpdb;
+
+		$cache_bucket = is_ssl() ? 'widget_authors_ssl' : 'widget_authors';
+
+		if ( '%BEG_OF_TITLE%' != $args['before_title'] ) {
+			if ( $output = wp_cache_get( $cache_bucket, 'widget') ) {
+				echo $output;
+				return;
+			}
+
+			ob_start();
+		}
+
+		$instance = wp_parse_args( $instance, array( 'title' => __( 'Authors', 'jetpack' ), 'all' => false, 'number' => 5, 'avatar_size' => 48 ) );
+		$instance['number'] = min( 10, max( 0, (int) $instance['number'] ) );
+
+		// We need to query at least one post to determine whether an author has written any posts or not
+		$query_number = max( $instance['number'], 1 );
+
+		$authors = get_users( array(
+			'fields' => 'all',
+			'who' => 'authors'
+		) );
+
+		echo $args['before_widget'];
+		echo $args['before_title'] . esc_html( $instance['title'] ) . $args['after_title'];
+		echo '<ul>';
+
+		foreach ( $authors as $author ) {
+			$r = new WP_Query( array(
```

Comment:
Maybe we could add a filter here as well, as some sites don't really uses posts, and would want to use the widget even though their site uses mostly Custom Post Types, for example.

This way we could even include our own Jetpack Custom Post Types in there.